operators/src/source/gdal_source/mod.rs

```diff
@@ -144,6 +161,95 @@ pub struct GdalDatasetGeoTransform {
     pub y_pixel_size: f64,
 }
 
+impl GdalDatasetGeoTransform {
+    pub fn standardized_spatial_partition(
+        &self,
+        x_size: usize,
+        y_size: usize,
+    ) -> SpatialPartition2D {
+        // The geo transform of a gdal dataset might have flipped axis. So correct that.
```

Comment:
also: can't you, instead, just compute the window without the epsilon and check if the resulting pixels fall out of the the partition and subtract one then?

Reply:
[2ca4d35](https://github.com/geo-engine/geoengine/pull/602/commits/2ca4d35f8820be9271bbe1b9f678efd207605762)

Reply:
[8fd3b20](https://github.com/geo-engine/geoengine/pull/602/commits/8fd3b20b83382d3b413047c666ec86f744e593b1)

Reply:
what do you think about the suggestion of having two separate data structures for SpatialPartitions? Or at least having a `is_standardized()` method and debug assert everywhere, where we expect it to not be flipped, that it isn't.

Reply:
the SpatialPartition is never flipped. As I see it the SpatialPartition does not know about directions. Only a GeoTransform can indicate the direction the pixels are stored. We have to sperate a few things here.
A GeoTransform translate from pixel space to world space. The pixel size sign (y) indicates how the raster rows are stored in your memory layout.
SRS coordinates in map space are usually arranged like your normal coordinate system. However, there might be the case that a SRS has an up -> down y-axis. Then our SpatialPartition will not work. Well, it will work but the edge at lower left is actually the edge with the smallest values. It will then be at the top-left and the SpatialPartition is up-side-down.